### ISSUE TITLE
[BUG] Crash on Map Load. Access Violation.

### ISSUE THREAD

**Issue by FDinolfo** [coelckers/gzdoom#1610]
### GZDoom version

4..7.1, 4.8.0 pre

### Which game are you running with GZDoom?

Other

### What Operating System are you using?

Windows 10

### Please describe your specific OS version

Windows 10Home / Windows 11 Home

### Relevant hardware info

In crash reports

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

Loading up our work in progress maps caused a crash. I was experimenting with adding Activation/DeActivation to an actor, which is where the problem may have stemmed from. The crash, though, seemed to be more frequent on the dev build.

On my friend's 4.7.1 instance, Maps 1, 2, 3, would all crash, but 4 would not. On my instance of 4.8.0 pre all four would crash. Map 4 didn't even have the actor I was messing with in it.

After reverting the changes I made to the actor I was working on, it started working on 4.7.1 again, but I've not gone back to the dev build to verify that it works there.

This may not be a particularly helpful report, but maybe there's something in the attachments that might shed some light. Especially if it can point out what it was that I did that it didn't liike.

### Steps to reproduce the behaviour.

Explain how to reproduce
1. Unpack the pk3 to a folder.
3. Open UDB and go to Game Configurations.
4. Add the Directory as a resource, make sure it's the last entry in the list.
5. Load the maps one at a time and press F9 in UDB to launch GZDoom.
6. It should crash.

 [CrashReport.zip](https://github.com/coelckers/gzdoom/files/8709283/CrashReport.zip)
[CrashReport2.zip](https://github.com/coelckers/gzdoom/files/8709284/CrashReport2.zip)

**Comment by coelckers** [comment]
All I can say here is "No map, no fix" I need to run this in the debugger to find out what's happening.

**Comment by FDinolfo** [comment]
We're talking about a half gigabyte file here. I can share it via dropbox, but was hesitant to because of the size. If that's not an issue, I'll zip it and send it.

**Comment by FDinolfo** [comment]
Here is the file as a pk3. I removed the UDB scratch files to save bandwidth. There are four maps. Map01 to 04, and a fifth wep_test map which is a map where we test ideas and things we might implement.

The troubled file is in the zscript/weapons folder, named dssecuritybot.zsc. I've also included the latest revision of that file with the extension .working if you want to switch them out to see changes.

These maps were tested with Project Brutality, but I tried it out with no mods and the crash is still there, so it's not a PB conflict.

I hope that this will be helpful, as I'd love to know why/how I managed to blow up the engine.

https://www.dropbox.com/s/3zqvc720eutqzwe/DragonSectorWIP.pk3?dl=0
